packages/commonwealth/client/scripts/views/pages/CommunityManagement/CommunityProfile/CommunityProfileForm/CommunityProfileForm.tsx

```diff
@@ -72,10 +65,71 @@ const CommunityProfileForm = () => {
       error: '',
     })),
   );
-  const { selectedTags, setSelectedTags, toggleTagFromSelection } =
-    usePreferenceTags({
-      initialSelectedTag: currentCommunityTags,
-    });
+
+  // TODO: this should be removed when /updateCommunityCategory route is updated to
+  // use only the newer tags schema
+  const [deprecatedCurrentCommunityTags, setDeprecatedCurrentCommunityTags] =
+    useState(
+      Object.entries(getCommunityTags(community.id))
+        .filter(({ 1: value }) => value)
+        .map(({ 0: key }, index) => ({ id: index, tag: key })),
+    );
+  const [deprecatedSelectedCommunityTags, setDeprecatedSelectedCommunityTags] =
+    useState(deprecatedCurrentCommunityTags);
+  const areTagsInitiallySet = useRef(false);
+  const {
+    isLoadingTags,
+    preferenceTags,
+    setPreferenceTags,
+    toggleTagFromSelection,
+  } = usePreferenceTags({});
+
+  const updatePreferenceTags = useCallback(() => {
+    const isOlderTagSelected = (tag: string, name: string) => {
+      return (
+        tag.toLowerCase() === name &&
+        !!deprecatedCurrentCommunityTags.find(
+          (dct) => dct.tag.toLowerCase() === name,
+        )
+      );
+    };
+
+    const updatedTags = [...preferenceTags].map((tag) => ({
+      ...tag,
+      isSelected:
+        // if `defi` or `dao` is in older tags, mark it selected in newer ones
+        isOlderTagSelected(tag.item.tag, 'defi') ||
+        isOlderTagSelected(tag.item.tag, 'dao') ||
+        !!(community.CommunityTags || []).find((t) => t.id === tag.item.id),
+    }));
+    setPreferenceTags(updatedTags);
```

Comment:
in my opinion this should be done as part of the db migration, not on the front end side

Reply:
Yup, its already planned for https://github.com/hicommonwealth/commonwealth/issues/7835